### PR TITLE
15.4 submission

### PR DESCRIPTION
**Pull request by xnox**
Please pulll tag ubuntu-shim-amd64+arm64-20210407 from my repo into CanonicalLtd/shim-review.

### CODE REVIEW

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1,24 +1,180 @@
...
+No, our grub enforces lockdown & uses shim protocol (rhboot linuxefi
+sb patches) to verify next component.
+
+###### What kernel are you using? Which patches does it includes to enforce Secure Boot?
+linux, various versions. They include lockdown pathces & ACPI patches,
+lockdown is enforced when booted with SecureBoot, config enforces
+kernel module signatures under lockdown.
+
+###### What changes were made since your SHIM was last signed?
+All patches dropped, as they were mrged upstream
```
Comment (dannf) [inline ISSUE_TEMPLATE.md]: s/mrged/merged/

File: ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -1,24 +1,180 @@
...
+Note that currently kernel does not yet implement correct checking of
+MokListXRT for kexec, thus one currently can kexec revoked/old kernels
+from the booted good kernel.
+
+###### Does your SHIM load any loaders that support loading unsigned kernels (e.g. GRUB)?
+No, our grub enforces lockdown & uses shim protocol (rhboot linuxefi
+sb patches) to verify next component.
+
+###### What kernel are you using? Which patches does it includes to enforce Secure Boot?
+linux, various versions. They include lockdown pathces & ACPI patches,
```
Comment (dannf) [inline ISSUE_TEMPLATE.md]: s/pathces/patches/

File: ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -1,24 +1,180 @@
...
+lockdown is enforced when booted with SecureBoot, config enforces
+kernel module signatures under lockdown.
+
+###### What changes were made since your SHIM was last signed?
+All patches dropped, as they were mrged upstream
+
+ - out of MokListRT mirror size considerations, we have stop using
```
Comment (dannf) [inline ISSUE_TEMPLATE.md]: s/stop/stopped/

File: ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -1,24 +1,180 @@
...
+   shim ephemeral signing certificate. And instead we use our signing
```
Comment (dannf) [inline ISSUE_TEMPLATE.md]: s/shim/a shim/ ?

File: README.md [hunk 5]
```diff
@@ -59,30 +65,113 @@ https://code.launchpad.net/~ubuntu-core-dev/shim/+git/shim/+ref/master
...
+https://git.launchpad.net/~ubuntu-core-dev/grub/+git/ubuntu/tree/debian/patches?h=ubuntu
+... and marked "linuxefi_*" for the SecureBoot specific patches.
+
+Patches from Debian are not specially marked; but as the code is based on Debian's GRUB,
+Ubuntu-specific patches are marked "ubuntu-*"
+
+Also note the many CVE patches in the same directory.
+
+-------------------------------------------------------------------------------
+What exact implementation of Secureboot in GRUB2 ( if this is your bootloader ) you have ?
```
Comment (dannf) [inline README.md:96]: s/you have/do you have/

File: README.md [hunk 6]
```diff
@@ -96,39 +185,60 @@ suffice.
...
+
+For our next shim after this one, we will ensure that kernel correctly
+rejects kernels for kexec based on MokListXRT including by-cert
+revocation, start signing such kernels with the new signing
+certificate, and replaces all the hashes in the vendor dbx with the
+currently in use certs.
+
+Other changes compared with our previous shim submissions:
+ - out of MokListRT mirror size considerations, we have stop using
+   shim ephemeral signing certificate. And instead we use our signing
```
Comment (dannf) [inline README.md]: s/shim/a shim/?

**Review by dannf — COMMENTED** [review]
A bunch of nits - feel free to ignore

File: ISSUE_TEMPLATE.md (hunk 4 above)
Comment (xnox) [reply ISSUE_TEMPLATE.md]: shim's

File: README.md (hunk 5 above)
Comment (xnox) [reply README.md:96]: this bit needs to go into upstream repo, as that's from them.

**Comment by xnox** [comment]
@dannf fixed up & retagged, if you are happy to pull this do pull the signed tag too.

**Pull request merged**